sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveExternalCatalogSuite.scala

```diff
@@ -154,4 +155,32 @@ class HiveExternalCatalogSuite extends ExternalCatalogSuite {
     catalog.createTable(hiveTable, ignoreIfExists = false)
     assert(catalog.getTable("db1", "spark_29498").owner === owner)
   }
+
+  test("SPARK-30868 throw QueryExectionException if run hive query failed") {
+    // test add jars which not exists
+    val jarPath = "file:///tmp/not_exists.jar"
+    assertThrows[QueryExecutionException](externalCatalog.client.runSqlHive(s"ADD JAR $jarPath"))
+
+    // test change to the database which not exists
+    val dbName = "db_not_exists"
+    assertThrows[QueryExecutionException](externalCatalog.client.runSqlHive(
+      s"use $dbName"))
+
+    // test create hive table failed
+    val tblName = "table_not_exists"
+    assertThrows[QueryExecutionException](externalCatalog.client.runSqlHive(
+      s"CREATE TABLE $tblName(n int)"))
```

Comment:
this should not fail

Reply:
maybe use a non existing data type?

Reply:
This is going to run in Hive, not in Spark. It is invalid in hive.
Or maybe I changed to a better query.